src/components/EditableHeading/EditableHeading.tsx

```diff
@@ -1,34 +1,67 @@
 /* eslint-disable react/destructuring-assignment */
 import React, { useCallback, useEffect, useLayoutEffect, useRef, useState } from "react";
-import PropTypes from "prop-types";
 import cx from "classnames";
-import Heading from "../Heading/Heading";
+import Heading, { HeadingProps } from "../Heading/Heading";
 import Clickable from "../Clickable/Clickable";
-import EditableInput from "../EditableInput/EditableInput";
+import EditableInput, { EditableInputProps } from "../EditableInput/EditableInput";
 import usePrevious from "../../hooks/usePrevious";
 import { InputType } from "../EditableInput/EditableInputConstants";
 import "./EditableHeading.scss";
-
-const EditableHeading = props => {
+import { HeadingSizes, HeadingTypes } from "../Heading/HeadingConstants";
+
+export interface EditableHeadingProps extends EditableInputProps, HeadingProps {
+  displayPlaceholderInTextMode?: boolean;
+  suggestEditOnHover?: boolean;
+  autoSize?: boolean;
+  inputAriaLabel?: string;
+  placeholder?: string;
+  errorClass?: string;
+  ignoreBlurClass?: string;
+  id?: string;
+  className?: string;
+  inputClassName?: string;
+  dataTestId?: string;
+  value?: string;
+  editing?: boolean;
+  disabled?: boolean;
+  errorClassTimeout?: number;
+  style?: React.CSSProperties;
+  customColor?: string;
+  onStartEditing?: (event: React.KeyboardEvent) => void;
+  tooltip?: string;
+  highlightTerm?: string;
+  insetFocus?: boolean;
+  contentRenderer?: React.FC;
+}
+
+const EditableHeading: React.FC<EditableHeadingProps> & {
+  sizes?: typeof HeadingSizes;
+  types?: typeof HeadingTypes;
+} = props => {
```

Comment:
maybe it'd better to deconstruct props here, as we always do?

Reply:
@SergeyRoyt 
we have tons of overrides with the same name
this component requires heavy refactoring - not in this PR though